core/core/src/ansible/roles/filebeat/tasks/configure-filebeat.yml

```diff
@@ -7,12 +7,26 @@
   set_fact: kibana_address="{{ hostvars[groups['kibana'][0]]['ansible_default_ipv4']['address'] }}"
   when: groups['kibana'][0] is defined
 
-- name: Copy configuration file
+- name: Copy configuration file (filebeat.yml)
   template:
     dest: /etc/filebeat/filebeat.yml
     src: filebeat.yml.j2
   notify: Restart Filebeat service
 
+- name: Set Filebeat to be started after Docker
+  block:
+    - name: Create directory (filebeat.service.d)
+      file:
+        path: /etc/systemd/system/filebeat.service.d
+        state: directory
+    - name: Copy drop-in configuration file (extra-dependencies.conf)
+      template:
+        dest: /etc/systemd/system/filebeat.service.d/extra-dependencies.conf
+        src: extra-dependencies.conf.j2
+      register: filebeat_unit_dependencies
+      notify: Restart Filebeat service
+  when: inventory_hostname in groups['master'] or inventory_hostname in groups['worker']
+
```

Comment:
This block should be placed before "Copy configuration file (filebeat.yml)" <- there's a handler for restarting Filebeat.

first change configuration
then enable and restart services

Reply:
Whoops. My bad, handlers are flushed later in main.yml. Looks good.

Reply:
That's ok since restart is triggered later, in main.yml by flush_handlers task.